marimo/_utils/distributor.py

```diff
@@ -49,9 +50,17 @@ def _remove() -> None:
 
     def _on_change(self) -> None:
         """Distribute the response to all consumers."""
+        retry_sleep_seconds = 0.001
         while self.input_connection.poll():
             try:
                 response = self.input_connection.recv()
+            except BlockingIOError as e:
+                # recv() sporadically fails with EAGAIN, EDEADLK ...
+                LOGGER.warning(
+                    "BlockingIOError in distributor receive: %s", str(e)
+                )
+                time.sleep(retry_sleep_seconds)
```

Comment:
Aren't hard sleeps blocking? Any reason not to use asyncio?

Reply:
They are blocking, but `asyncio.add_reader()` doesn't accept an async callback. So we can't use `asyncio.sleep()` here.

Reply:
It might even be possible to remove the sleep? 1000ns is just an order magnitude larger than a function call, and might spend that on Logging

You can still invoke asyncio.run(asyncio sleep(time)) from a sync call

Reply:
> You can still invoke asyncio.run(asyncio sleep(time)) from a sync call

No that's not possible here; an event loop is already running. That would also be blocking, even if it were possible :)